crates/host/src/wasmbus/mod.rs

```diff
@@ -978,14 +985,79 @@ impl Host {
             ctl_nats_url = config.ctl_nats_url.as_str(),
             "connecting to NATS control server"
         );
-        let nats = async_nats::connect(config.ctl_nats_url.as_str())
-            .await
-            .context("failed to connect to NATS control server")?;
+        let ctl_nats = match (config.ctl_jwt.as_ref(), config.ctl_seed.as_ref()) {
+            (Some(jwt), Some(seed)) => {
+                let kp = std::sync::Arc::new(
+                    nkeys::KeyPair::from_seed(seed)
+                        .context("failed to construct key pair from seed")?,
+                );
+                async_nats::ConnectOptions::with_jwt(jwt.to_string(), move |nonce| {
+                    let key_pair = kp.clone();
+                    async move { key_pair.sign(&nonce).map_err(async_nats::AuthError::new) }
+                })
+            }
+            _ => async_nats::ConnectOptions::new(),
+        }
+        .require_tls(config.ctl_tls)
+        .connect(config.ctl_nats_url.as_str())
+        .await
+        .context("failed to connect to NATS control server")?;
+
+        let queue = Queue::new(
+            &ctl_nats,
+            &config.ctl_topic_prefix,
+            &config.lattice_prefix,
+            &cluster_key,
+            &host_key,
+        )
+        .await
+        .context("failed to initialize queue")?;
+        ctl_nats.flush().await.context("failed to flush")?;
```

Comment:
Nit: Might want to note why we are flushing here for future maintainers and future selves (I know this was already here, but hence the nit)

Reply:
Yeah I'm not actually sure why, but happy to add a comment if there's reasoning